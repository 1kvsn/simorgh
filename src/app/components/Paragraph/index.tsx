/** @jsx jsx */

import React, { FC } from 'react';
import { jsx } from '@emotion/react';

import { GelFontSize } from '../../interfaces';
import Text from '../Text';
import styles from './index.styled';

interface Props {
  className?: string;
  children?: React.ReactNode;
  size?: GelFontSize;
}

const Paragraph: FC<Props> = ({
  children,
  className,
  size,
  ...htmlAttributes
}: Props) => (
  <Text
    as="p"
    className={className}
    css={styles.paragraph}
    size={size || 'bodyCopy'}
    {...htmlAttributes}
  >
    {children}
  </Text>
);

export default Paragraph;
