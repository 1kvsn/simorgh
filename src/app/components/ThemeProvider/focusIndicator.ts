import { css, Theme } from '@emotion/react';
import pixelsToRem from '../../utilities/pixelsToRem';

const focusIndicator = ({ palette }: Theme) => css`
  // default styling
  a:focus {
    outline: ${pixelsToRem(3)}rem solid ${palette.BLACK};
    box-shadow: 0 0 0 ${pixelsToRem(3)}rem ${palette.WHITE};
    outline-offset: ${pixelsToRem(3)}rem;
  }

  a:focus:not(:focus-visible) {
    outline: none;
    box-shadow: none;
    outline-offset: 0;
  }

  a:focus-visible {
    outline: ${pixelsToRem(3)}rem solid ${palette.BLACK};
    box-shadow: 0 0 0 ${pixelsToRem(3)}rem ${palette.WHITE};
    outline-offset: ${pixelsToRem(3)}rem;
  }

  // add block display default styling
  a.focusIndicatorDisplayBlock:focus-visible {
    display: block;
  }

  a.someClassName:focus-visible {
    background-color: yellow;
  }
`;

export default focusIndicator;
