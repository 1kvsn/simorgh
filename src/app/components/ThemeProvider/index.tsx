import loadable, { LoadableComponent } from '@loadable/component';

interface Props {
  service:
    | 'afaanoromoo'
    | 'afrique'
    | 'amharic'
    | 'arabic'
    | 'archive'
    | 'azeri'
    | 'bengali'
    | 'burmese'
    | 'cymrufyw'
    | 'gahuza'
    | 'gujarati'
    | 'hausa'
    | 'hindi'
    | 'igbo'
    | 'indonesia'
    | 'japanese'
    | 'korean'
    | 'kyrgyz'
    | 'marathi'
    | 'mundo'
    | 'naidheachdan'
    | 'nepali'
    | 'news'
    | 'newsround'
    | 'pashto'
    | 'persian'
    | 'pidgin'
    | 'portuguese'
    | 'punjabi'
    | 'russian'
    | 'scotland'
    | 'serbian'
    | 'sinhala'
    | 'somali'
    | 'sport'
    | 'swahili'
    | 'tamil'
    | 'telugu'
    | 'thai'
    | 'tigrinya'
    | 'turkce'
    | 'ukchina'
    | 'ukrainian'
    | 'urdu'
    | 'uzbek'
    | 'vietnamese'
    | 'yoruba'
    | 'zhongwen';
  variant: null | 'cyr' | 'lat' | 'simp' | 'trad';
}

// This is to support local 404 and 500 pages for services with variants.
// For example, http://localhost:7080/serbian/500 does not include the variant in the path so we must define a default variant.
// NB These routes do not exist on live. They are for dev and testing purposes.
const defaultServiceVariants: { [index: string]: any } = {
  serbian: 'cyr',
  ukchina: 'simp',
  zhongwen: 'simp',
};

const loadTheme = /* #__LOADABLE__ */ (props: Props) =>
  import(
    `./themes/${props.service}${
      props.variant
        ? `/${props.variant}`
        : `/${defaultServiceVariants[props.service]}` || ''
    }`
  );

const ThemeProvider: LoadableComponent<Props> = loadable(loadTheme);

export default ThemeProvider;
