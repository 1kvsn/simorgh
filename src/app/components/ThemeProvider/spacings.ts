import pixelsToRem from '../../utilities/pixelsToRem';

export const FULL = pixelsToRem(16);
export const DOUBLE = FULL * 2;
export const TRIPLE = FULL * 3;
export const QUADRUPLE = FULL * 4;
