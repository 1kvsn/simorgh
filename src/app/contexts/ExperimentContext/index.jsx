import React, { createContext } from 'react';


const ExperimentContext = React.createContext({});

const ExperimentContextProvider = ({ children, dataMvtValue }) => {
  return (
    <ExperimentContext.Provider value={dataMvtValue}>
      {children}
    </ExperimentContext.Provider>
  );
};

const ExperimentContextConsumer = ExperimentContext.Consumer;

export { ExperimentContext, ExperimentContextProvider, ExperimentContextConsumer };
