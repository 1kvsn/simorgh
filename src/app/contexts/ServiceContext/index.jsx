import React from 'react';
import { node, string } from 'prop-types';
import services from '#lib/config/services/loadableConfig';
import variantPropType from '../../models/propTypes/variants';
import { getVariant } from '#lib/utilities/variantHandler';
import getLangOverride from '#lib/utilities/langHandler';

export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({
  children,
  service,
  variant,
  pageLang,
}) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={getLangOverride(pageLang) || getVariant({ service, variant })}
    >
      {children}
    </LoadableContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
  variant: variantPropType,
  pageLang: string,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
  variant: 'default',
  pageLang: null,
};
