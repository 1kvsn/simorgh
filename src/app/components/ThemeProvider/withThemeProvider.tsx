import { Global, ThemeProvider as EmotionThemeProvider } from '@emotion/react';
import mergeDeepLeft from 'ramda/src/mergeDeepLeft';

import * as colours from './colours';
import * as mq from './mediaQueries';
import * as spacings from './spacings';

import { BrandColours, Typography } from '../../interfaces';

interface ServiceTheme {
  colours: BrandColours;
  typography: Typography;
}

type Props = {
  children: JSX.Element | JSX.Element[];
};

const withThemeProvider = (serviceTheme: ServiceTheme) => {
  const ThemeProvider: React.FC<Props> = ({ children }: Props) => (
    <EmotionThemeProvider
      theme={mergeDeepLeft(serviceTheme, {
        colours,
        mq,
        spacings,
      })}
    >
      <Global styles={({ typography }) => typography.fontFaces} />
      {children}
    </EmotionThemeProvider>
  );

  return ThemeProvider;
};

export default withThemeProvider;
