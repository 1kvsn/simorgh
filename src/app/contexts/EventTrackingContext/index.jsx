import React, { createContext, useContext } from 'react';
import { node } from 'prop-types';

import { pageDataPropType } from '#models/propTypes/data';
import { ServiceContext } from '#contexts/ServiceContext';
import { RequestContext } from '#contexts/RequestContext';
import { buildATIClickParams } from '#containers/ATIAnalytics/params';
import useToggle from '#hooks/useToggle';
import {
  ARTICLE_PAGE,
  FRONT_PAGE,
  MEDIA_PAGE,
  MOST_READ_PAGE,
  MOST_WATCHED_PAGE,
  INDEX_PAGE,
  FEATURE_INDEX_PAGE,
  MEDIA_ASSET_PAGE,
  STORY_PAGE,
  PHOTO_GALLERY_PAGE,
  CORRESPONDENT_STORY_PAGE,
} from '#app/routes/utils/pageTypes';

export const EventTrackingContext = createContext({});

export const EventTrackingContextProvider = ({ children, pageData }) => {
  let pageIdentifier;
  let platform;
  let statsDestination;

  const requestContext = useContext(RequestContext);
  const serviceContext = useContext(ServiceContext);
  const { enabled: eventTrackingIsEnabled } = useToggle('eventTracking');

  if (pageData && eventTrackingIsEnabled) {
    try {
      ({ pageIdentifier, platform, statsDestination } = buildATIClickParams(
        pageData,
        requestContext,
        serviceContext,
      ));
    } catch (error) {
      // eslint-disable-next-line no-console
      console.error(
        `ATI Event Tracking Error: Could not parse tracking values from page data:\n${error.message}`,
      );
    }
  }

  const { pageType } = requestContext;
  const campaignName = {
    [ARTICLE_PAGE]: 'article',
    [FRONT_PAGE]: 'index-home',
    [MEDIA_PAGE]: 'player-episode-tv',
    [MOST_READ_PAGE]: 'list-datadriven-read',
    [MOST_WATCHED_PAGE]: 'list-datadriven-watched',
    [INDEX_PAGE]: 'index-section',
    [FEATURE_INDEX_PAGE]: 'index-section-fix',
    [MEDIA_ASSET_PAGE]: 'article-media-asset',
    [STORY_PAGE]: 'article-sty',
    [PHOTO_GALLERY_PAGE]: 'article-photo-gallery',
    [CORRESPONDENT_STORY_PAGE]: 'article-csp',
  }[pageType];

  if (!campaignName && pageData) {
    // eslint-disable-next-line no-console
    console.error(
      `ATI Event Tracking Error: Could not get the page type's campaign name`,
    );
  }

  const hasRequiredProps = [
    campaignName,
    pageIdentifier,
    platform,
    statsDestination,
  ].every(Boolean);
  const trackingData = hasRequiredProps
    ? {
        campaignName,
        pageIdentifier,
        platform,
        statsDestination,
      }
    : {};

  return (
    <EventTrackingContext.Provider value={trackingData}>
      {children}
    </EventTrackingContext.Provider>
  );
};

EventTrackingContextProvider.propTypes = {
  children: node.isRequired,
  pageData: pageDataPropType,
};

EventTrackingContextProvider.defaultProps = {
  pageData: null,
};
