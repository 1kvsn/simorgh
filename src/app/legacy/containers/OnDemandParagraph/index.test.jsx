import React from 'react';
import {
  shouldMatchSnapshot,
  isNull,
  suppressPropWarnings,
} from '#psammead/psammead-test-helpers/src';
import { ServiceContextProvider } from '../../../contexts/ServiceContext';
import ThemeProvider from '../../../components/ThemeProvider';
import Paragraph from '.';

jest.mock('../../../components/ThemeProvider');

describe('MediaPageBlocks Paragraph', () => {
  shouldMatchSnapshot(
    'should render correctly',
    <ServiceContextProvider service="news">
      <ThemeProvider service="news" variant="default">
        <Paragraph uuid="uuid" idAttr="idAttr" text="Example text" />
      </ThemeProvider>
    </ServiceContextProvider>,
  );

  shouldMatchSnapshot(
    'should render correctly - dark mode',
    <ServiceContextProvider service="news">
      <ThemeProvider service="news" variant="default">
        <Paragraph uuid="uuid" idAttr="idAttr" text="Example text" darkMode />
      </ThemeProvider>
    </ServiceContextProvider>,
  );

  describe("when text isn't provided", () => {
    suppressPropWarnings(['text', 'undefined']);

    isNull(
      'should render null',
      <ServiceContextProvider service="news">
        <ThemeProvider service="news" variant="default">
          <Paragraph uuid="uuid" idAttr="idAttr" />
        </ThemeProvider>
      </ServiceContextProvider>,
    );
  });
});
