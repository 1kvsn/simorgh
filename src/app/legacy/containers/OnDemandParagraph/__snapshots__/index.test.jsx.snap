// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPageBlocks Paragraph should render correctly - dark mode 1`] = `
.emotion-1 {
  color: #141414;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 400;
  margin: 0;
  padding-bottom: 16px;
  color: #F2F2F2;
}

@media (min-width: 20rem) {
  .emotion-1 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-1 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (max-width: 62.9375rem) {
  .emotion-1 {
    padding-bottom: 0.5rem;
  }
}

<div>
  <p
    class="emotion-0 emotion-1"
    id="idAttr"
  >
    Example text
  </p>
</div>
`;

exports[`MediaPageBlocks Paragraph should render correctly 1`] = `
.emotion-1 {
  color: #141414;
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-style: normal;
  font-weight: 400;
  margin: 0;
  padding-bottom: 16px;
  color: #6E6E73;
}

@media (min-width: 20rem) {
  .emotion-1 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-1 {
    font-size: 1rem;
    line-height: 1.375rem;
  }
}

@media (max-width: 62.9375rem) {
  .emotion-1 {
    padding-bottom: 0.5rem;
  }
}

<div>
  <p
    class="emotion-0 emotion-1"
    id="idAttr"
  >
    Example text
  </p>
</div>
`;
